usaspending_api/download/management/commands/populate_monthly_delta_files.py

```diff
@@ -0,0 +1,189 @@
+        raw_quoted_query = generate_raw_quoted_query(source_query)
+        csv_query_annotated = self.apply_annotations_to_sql(raw_quoted_query, source.human_names)
+        (temp_sql_file, temp_sql_file_path) = tempfile.mkstemp(prefix='bd_sql_', dir='/tmp')
+        with open(temp_sql_file_path, 'w') as file:
+            file.write('\\copy ({}) To STDOUT with CSV HEADER'.format(csv_query_annotated))
+
+        # Generate the csv with \copy
+        cat_command = subprocess.Popen(['cat', temp_sql_file_path], stdout=subprocess.PIPE)
+        subprocess.check_output(['psql', '-o', source_path, os.environ['DOWNLOAD_DATABASE_URL'], '-v',
+                                 'ON_ERROR_STOP=1'], stdin=cat_command.stdout, stderr=subprocess.STDOUT)
+
+        # Split CSV into separate files
+        split_csvs = split_csv(source_path, row_limit=EXCEL_ROW_LIMIT, output_path=os.path.dirname(source_path),
+                               output_name_template='{}_delta_%s.csv'.format(source_name))
+
+        # Zip the split CSVs into one zipfile
+        zipfile_path = '{}{}_{}_Delta_{}.zip'.format(settings.CSV_LOCAL_PATH, agency_code, award_type,
+                                                     datetime.strftime(date.today(), '%Y%m%d'))
+        zipped_csvs = zipfile.ZipFile(zipfile_path, 'a', compression=zipfile.ZIP_DEFLATED, allowZip64=True)
+        for split_csv_part in split_csvs:
+            zipped_csvs.write(split_csv_part, os.path.basename(split_csv_part))
+
+        os.close(temp_sql_file)
+        os.remove(temp_sql_file_path)
+        shutil.rmtree(working_dir)
+
+        return zipfile_path
+
+    def apply_annotations_to_sql(self, raw_query, aliases):
+        """The csv_generation version of this function would incorrectly annotate the D1 correction_delete_ind.
+        Reusing the code but including the correction_delete_ind col for both file types"""
+        select_string = re.findall('SELECT (.*?) FROM', raw_query)[0]
+
+        selects, cases = [], []
+        for select in select_string.split(','):
+            if 'CASE' not in select:
+                selects.append(select.strip())
+            else:
+                case = select.strip()
+                cases.append(case)
+                aliases.remove(re.findall('AS "(.*?)"', case)[0].strip())
+        if len(selects) != len(aliases):
+            raise Exception("Length of alises doesn't match the columns in selects")
+
+        ordered_map = OrderedDict(zip(aliases, selects))
+        selects_str = ", ".join(cases + ['{} AS \"{}\"'.format(select, alias) for alias, select in ordered_map.items()])
+
+        return raw_query.replace(select_string, selects_str)
+
+    def parse_filters(self, award_types, agency):
+        """Convert readable filters to a filter object usable for the matview filter"""
+        award_type_codes = []
+        for award_type in award_types:
+            award_type_codes = award_type_codes + all_award_types_mappings[award_type]
```

Comment:
More of a code style preference, your call:
` award_type_codes = [all_award_types_mappings[award_type] for award_type in award_types]`